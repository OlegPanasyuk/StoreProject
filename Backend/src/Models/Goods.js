module.exports = (sequalize, type) => {
    return sequalize.define( 'goods', {
        idgoods : {
            type : type.INTEGER,
            primaryKey : true,
            autoIncrement : true
        },
        name : {
            type : type.STRING(200),
            validate : {
                allowNull : false
            }
        },
        description : {
            type : type.TEXT   
        }
    }, {
        timestamps : false,
        freezeTableName : true, 
        tableName : 'goods'
    });
};