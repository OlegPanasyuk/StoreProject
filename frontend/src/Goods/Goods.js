import React, { Component } from 'react';
import { Card, CardColumns } from 'react-bootstrap';
import PropTypes from 'prop-types';

class Goods extends Component {
    constructor(props) {
        super(props);
        this.state = {

        };
    }

    render() {
        let { obj } = this.props;
        
        let itemsGoods = Object.keys(obj).map((el, i)=>{
            return (
                <Card key={`Good-${obj[el].idgoods}-${i}`}>
                    <Card.Header>
                        {obj[el].name}
                    </Card.Header>
                    <Card.Text>
                        {obj[el].description}
                    </Card.Text>
                </Card>
            );
        });
        if (itemsGoods.length === 0) {
            itemsGoods = 'There no any goods in categoty or select another item in menu';
        }
        return (
            <CardColumns style={{columnGap: '1em'}}>
                {itemsGoods}
            </CardColumns>
        );
    }
}

Goods.propTypes = {
    obj : PropTypes.array
};

export default Goods;