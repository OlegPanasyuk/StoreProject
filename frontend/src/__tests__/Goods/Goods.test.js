import React from 'react';
import Goods from '../../Goods/Goods';
import {shallow, mount, configure } from 'enzyme';
import {shallowToJson,  mountToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    obj : [{
        idgoods : 1,
        name : 'Honda Civic',
        description : 'Lorem ipsum dolor sit amet consectetur adipisicing elit. \
        Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates \
        quis voluptate, dolorem modi non? Nisi, maxime a.',
        catalogue_id_catalogue : 4
    },{
        idgoods : 2,
        name : 'Volvo',
        description : 'Lorem ipsum dolor sit amet consectetur adipisicing elit. \
        Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates \
        quis voluptate, dolorem modi non? Nisi, maxime a.',
        catalogue_id_catalogue : 4
    }]
}) {
    let component = shallow(<Goods {...props} />);
    return { component, props };
}


describe('Goods Component', () => {
    it('Component must be created and props must be correct', () => {
        const { component } = setupComponent();
        
        expect(Goods.propTypes.obj).toBe(PropTypes.array);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
    
});