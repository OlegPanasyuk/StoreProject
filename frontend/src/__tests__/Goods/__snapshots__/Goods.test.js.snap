// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goods Component Component must be created and props must be correct 1`] = `
<CardColumns
  style={
    Object {
      "columnGap": "1em",
    }
  }
>
  <Card
    as="div"
    body={false}
    key="Good-1-0"
  >
    <CardHeader>
      Honda Civic
    </CardHeader>
    <CardText>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.         Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates         quis voluptate, dolorem modi non? Nisi, maxime a.
    </CardText>
  </Card>
  <Card
    as="div"
    body={false}
    key="Good-2-1"
  >
    <CardHeader>
      Volvo
    </CardHeader>
    <CardText>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.         Blanditiis provident quo quaerat repellat fuga debitis nemo voluptates         quis voluptate, dolorem modi non? Nisi, maxime a.
    </CardText>
  </Card>
</CardColumns>
`;
